src/client/opamAction.ml

```diff
@@ -245,11 +245,12 @@ let download_package st nv =
   let dir = OpamSwitchState.source_dir st nv in
   if OpamPackage.Set.mem nv st.pinned &&
      OpamFilename.exists_dir dir &&
-     OpamStd.Option.Op.(
-       OpamPinned.find_opam_file_in_source nv.name dir >>|
-       OpamFile.OPAM.safe_read >>=
-       OpamFile.OPAM.version_opt)
-     = Some nv.version
+     (OpamSwitchState.is_version_pinned st (OpamPackage.name nv) ||
```

Comment:
Does this also return `true` when doing `opam pin add pkg.version <repo>`? In that case, there is no need to look into the archive/repo either